src/AppBundle/EventListener/RedirectToPreferredLocaleListener.php

```diff
@@ -81,6 +81,11 @@ public function onKernelRequest(GetResponseEvent $event)
         if (!$event->isMasterRequest() || '/' !== $request->getPathInfo()) {
             return;
         }
+        // Ignore requests from referrers with the same HTTP host in order to prevent
+        // changing language for users who possibly already selected it for this application.
+        if (0 === stripos($request->server->get('HTTP_REFERER'), $request->getSchemeAndHttpHost())) {
```

Comment:
@bocharsky-bw, it was a joke. I just meant that using `$request->headers->get('REFERER')` is more appropriate for me.